### PR TITLE
Fixes #204: Update backend to use logger

### LINKED ISSUE #204: Update Backend to Use Logger
#142 has landed and we can start logging instead of calling ```console.log```. We need to update the existing backend code.

### PR DESCRIPTION
**Pull request by kartik-budhiraja**
Fixes #204 , Updated backend to use logger

### CODE REVIEW

**Comment by miggs125** [comment]
@humphd @kartik-budhiraja there is a lot of code inside src/ that belongs inside src/backend/, such as feed*js Should we move this code to that directory, change the console logs there (if any) and then merge this code along with any additional changes that make use of the logger?

File: src/index.js [hunk 1]
```diff
@@ -5,7 +5,9 @@
 const fs = require('fs');
 const feedQueue = require('./feed-queue');
 const feedWorker = require('./feed-worker');
+const { logger } = require('./lib/logger');
 
+const log = logger.child({ module: 'basic-queue' });
```
Comment (ragnarokatz) [inline src/index.js]: The naming for the child logger should be consistent with other usages. In this case, it should be `{module: 'index'}`, for `index.js`.

**Comment by miggs125** [comment]
@kartik-budhiraja I noticed you actually got all the ``` console.log ``` calls. Great job.

File: src/index.js (hunk 1 above)
Comment (kartik-budhiraja) [reply src/index.js]: Module as `index` will be pretty vague to understand in the logs, and this index.js file is something to get us started for the queue handling, that's why logs with the module name as `basic-queue` make more sense to me.

**Comment by Reza-Rajabi** [comment]
Today, I made a stupid judgment on your code. I thought it is soon to change all `console.log` s to use logger because I didn't notice that logger has been set to use environment variables. Sorry if I caused confusion for anybody.

**Comment by humphd** [comment]
This needs a rebase for changes to the Passport logging that just landed.

**Comment by humphd** [comment]
More updates required to get this into shape after @miggs125's #350 

I'd love to get this landed soon so we can start using it.

**Comment by humphd** [comment]
@kartik-budhiraja how did you do your rebase?  You shouldn't have all these other commits from master in your branch.

**Comment by kartik-budhiraja** [comment]
@humphd  That's exactly what I am looking into rn, I guess I messed it up somewhere.
The flow I followed was:
- `git rebase master`
- Solved the merge conflicts
- `git rebase --continue`
- `git pull`
- `git push`

**Comment by humphd** [comment]
`git pull` is always going to merge after it fetches.  This is often what messes people up:

```
git checkout master
git pull upstream master
git checkout your-issue-branch
git rebase master
solve conflicts
git rebase --continue
git push your-issue-branch -f
```

**Comment by kartik-budhiraja** [comment]
@humphd True, sorry about this. The branch is rebased now properly.

File: src/backend/login/usingPassport.js [hunk 2]
```diff
@@ -28,6 +31,11 @@ passport.use(
     },
     // This function returns 'issuer:' issuer, 'userId:' userId, 'accessToken:' accessToken, 'refreshToken:' refreshToken, 'params:' params
     (req, issuer, userId, profile, accessToken, refreshToken, params, callback) => {
+      log.info('issuer:', issuer);
+      log.info('userId:', userId);
+      log.info('accessToken:', accessToken);
```
Comment (humphd) [inline src/backend/login/usingPassport.js]: This is re-adding something we removed in a recent commit.  We don't want to print the accessToken or refreshToken I don't think, cc @Grommers00 to confirm.

Comment (kartik-budhiraja) [reply src/backend/login/usingPassport.js]: Oh okay, I added it back during rebase, did not know it was removed on purpose, will remove it.

**Pull request merged**